### PR TITLE
DEC-1177 Add media type to index and reindex of item

### PR DESCRIPTION
**Pull request by rfox2**
We needed to have an item reindexed when a thumbnail was added so that it would appear in the item API output, and the media type is now output in search results. Some of the index classes were touched, and the item reindex call was added to the media thumbnail class.

### CODE REVIEW

File: app/models/waggle/adapters/solr/search/hit.rb [hunk 1]
```diff
@@ -46,6 +46,10 @@ def description
             fetch_text(:description)
           end
 
+          def media
+            eval(fetch_string(:media))
```
Comment (danthewolfe) [inline app/models/waggle/adapters/solr/search/hit.rb]: Sounds risky.

**Comment by danthewolfe** [comment]
:+1:

File: app/models/waggle/adapters/solr/search/hit.rb (hunk 1 above)
Comment (jgondron) [inline app/models/waggle/adapters/solr/search/hit.rb]: Is the use of eval needed? We should also handle this somewhere else, such as whatever class is responsible for mapping a Honeycomb item to a Waggle item.

File: app/models/waggle/item.rb [hunk 2]
```diff
@@ -29,7 +28,11 @@ def collection_id
     end
 
     def type
-      TYPE
+      if media
+        media["@type"]
+      else
+        "Unknown"
```
Comment (jgondron) [inline app/models/waggle/item.rb]: I think Jon mentioned actually having a "Metadata only" type item in this case. Idk what value we use, but something along those lines. Maybe we just fall back to "Item"?

**Comment by jgondron** [comment]
👍  Once the metadata only type is addressed.

**Pull request merged**